.github/workflows/build-test-distribute.yaml

```diff
@@ -170,3 +173,153 @@ jobs:
           trap on_exit EXIT
           make docker/push
           make docker/manifest
+  gen_e2e_matrix:
+    runs-on: ubuntu-latest
+    needs: ["build"]
+    outputs:
+      slow: ${{ steps.generate-matrix.outputs.slow }}
+      main: ${{ steps.generate-matrix.outputs.main }}
+      delta_kds: ${{ steps.generate-matrix.outputs.delta_kds }}
+      cni: ${{ steps.generate-matrix.outputs.cni }}
+    steps:
+      - name: Install jq
+        run: |
+          sudo apt-get install -y jq
+      - id: generate-matrix
+        name: Generate matrix
+        run: |
+          SKIP_CI_LABEL="${{ contains(github.event.pull_request.labels.*.name, 'ci/skip-test') || contains(github.event.pull_request.labels.*.name, 'ci/skip-e2e-test')  }}"
+          if [[ "$SKIP_CI_LABEL" == "true" ]]; then
+            echo "Skipping e2e jobs because the pull request contained a 'ci/skip-*' label."
+            exit 0
+          fi
+
+          JQ_PATH=$(which jq)
+          if [ "$JQ_PATH" == "" ]; then
+            echo "Could not find devtool 'jq', please check if 'check' job has run correctly prior to this job."
+            exit 1
+          fi
```

Comment:
Would it be more readable to have the full matrix and then a bunch of exclude? Something like:

```
BASE_MATRIX=<<<EOF
{
   "target": ["test/e2e", "test/e2e-universal", "test/e2e-kubernetes", "test/e2e-multizone"],
   "cniNetworkPlugin": ["flannel"],
   "arch": ["amd64", "arm64"],
   "paralellism": ["1"],
  "k8sVersion": ["kind", "kindIpv6", "${{ env.K8S_MIN_VERSION }}", "${{ env.K8S_MAX_VERSION }}"],
  // exclude all the invalid options
  "include":[
    {"target": "test/e2e", "paralellism": "4"},
    {"target": "test/e2e-kubernetes", "cniNetworkPlugin": "calico"},
  ],
  "exclude": [
    {"target": "test/e2e", "paralellism": "1"},
    {"target": "test/e2e-universal",  "${{ env.K8S_MIN_VERSION }}"},
    {"target": "test/e2e-universal", "${{ env.K8S_MAX_VERSION }}"}
  ]
}
EOF

EXCLUDE_OPTIONS=<<<EOF
[
   {"arch": "arm64},
   {"k8sVersion": "kindIpv6"},
   {"k8sVersion": "${{ env.K8S_MIN_VERSION}}"},
   {"cniNetworkPlugin": "calico"}
]
EOF

if [[ "$RUN_FULL_MATRIX" == "false" ]]; then
   BASE_MATRIX=$JQ_PATH -rc ".exclude += '${EXCLUDE_OPTIONS}'"
fi
```

There are probably mistake in the above but that's the idea.
I think this would help on readability and make this matrix more maintainable in the future WDYT?